### PR TITLE
Notaries Marketplace

### PR DESCRIPTION
**Pull request by andregoncalves**

### CODE REVIEW

File: src/renderer/marketplace/notarization/list/notarization-offers-container.jsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	const notaries = marketplaceSelectors.selectNotaries(state, identity.type);
+	// const vendors = marketplaceSelectors.selectVendorsForCategory(state, 'notaries');
+
+	// Find country from identity attributes
+	// Select US or international product
+	const countryAttribute = profile.allAttributes.find(
+		attr =>
+			attr.type.content.$id ===
+			'http://platform.selfkey.org/schema/attribute/nationality.json'
+	);
```
Comment (sk91) [inline src/renderer/marketplace/notarization/list/notarization-offers-container.jsx:51]: complex state to props, probably some selectors are missing

File: src/renderer/selfkey-id/main/components/message-container.jsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+const mapStateToProps = (state, props) => {
+	return {};
+};
+
+const styledComponent = withStyles(styles)(MessageContainer);
+const connectedComponent = connect(mapStateToProps)(styledComponent);
+export default connectedComponent;
+export { connectedComponent as MessageContainer };
+
+// 	{
```
Comment (sk91) [inline src/renderer/selfkey-id/main/components/message-container.jsx:24]: clean please

File: src/common/kyc/index.js [hunk 3]
```diff
@@ -323,8 +328,8 @@ export const kycSelectors = {
 	selectRequirementsForTemplate(state, rpName, templateId, identityId) {
 		const template = this.oneTemplateSelector(state, rpName, templateId);
 		if (!template) {
-			log.error(
-				`Unable to select template ${templateId} for ${rpName}, check if instance contains this template ID`
+			log.warn(
```
[the remark this answers is not in the document]
Comment (sk91) [reply src/common/kyc/index.js:331]: > Can we have a loading indicator instead?

I mean correct flag in state like "loading"

[the remark this answers is not in the document]
Comment (andregoncalves) [reply src/common/kyc/index.js]: This for debugging only. Changed to log only in dev.

File: src/main/marketplace/notaries-service.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+import request from 'request';
+import { Logger } from 'common/logger';
+import config from 'common/config';
+
+const log = new Logger('marketplace-notaries-service');
+
+const URL = config.notariesApiUrl;
+
+export class NotariesService {
+	loadNotaries() {
```
[the remark this answers is not in the document]
Comment (andregoncalves) [reply src/main/marketplace/notaries-service.js]: Nice catch. This is old code from initial certifiers branch work last year, removed it.

File: src/common/notaries/operations.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+import { createAliasedAction } from 'electron-redux';
+import { getGlobalContext } from 'common/context';
+import { notariesActions } from './actions';
+import { notariesTypes } from './types';
+
+const loadNotariesOperation = () => async (dispatch, getState) => {
+	await dispatch(notariesActions.setLoadingAction(true));
+	await dispatch(notariesActions.setErrorAction(null));
```
[the remark this answers is not in the document]
Comment (andregoncalves) [reply src/common/notaries/operations.js]: This is old code from initial certifiers branch work last year, removed it.

File: src/common/notaries/operations.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+	try {
+		const data = await getGlobalContext().notariesService.loadNotaries();
+		await dispatch(notariesActions.setMainAction(data.main));
+		await dispatch(notariesActions.setJurisdictionsAction(data.jurisdictions));
+		await dispatch(notariesActions.setDetailsAction(data.details));
+	} catch (error) {
+		console.error(error);
+		await dispatch(notariesActions.setErrorAction(true));
```
[the remark this answers is not in the document]
Comment (andregoncalves) [reply src/common/notaries/operations.js]: This is old code from initial certifiers branch work last year, removed it.

File: src/renderer/marketplace/notarization/list/notarization-offers-container.jsx (hunk 1 above)
Comment (andregoncalves) [reply src/renderer/marketplace/notarization/list/notarization-offers-container.jsx]: Yeah, I should be using existing selectors instead of this logic.

**Pull request merged**